Documentation/design-docs/first-class-structs.md

```diff
@@ -0,0 +1,608 @@
+   │  ┌──▌  const     struct 0
+   └──▌  =         struct (init)
+      └──▌  lclVar    struct V01 tmp0         
+
+   ▌  stmtExpr  void  (top level) (IL 0x008...  ???)
+   │  ┌──▌  const     struct 0
+   └──▌  =         struct (init)
+      └──▌  lclVar    struct V02 tmp1         
+
+   ▌  stmtExpr  void  (top level) (IL 0x008...  ???)
+   │  ┌──▌  const     struct 0
+   └──▌  =         struct (init)
+      └──▌  lclVar    struct V03 tmp2         
+
+   ▌  stmtExpr  void  (top level) (IL 0x008...  ???)
+   └──▌  call help long   HELPER.CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE
+      ├──▌  const     long   0x7ffc8bbb4e10
+      └──▌  const     int    1
+
+   ▌  stmtExpr  void  (top level) (IL 0x008...  ???)
+   │  ┌──▌  const     struct 0
+   └──▌  =         struct (init)
+      └──▌  obj(16)   struct
+         │  ┌──▌  const     long   8 Fseq[#FirstElem]
+         └──▌  +         byref 
+            └──▌  indir     ref   
+               └──▌  const(h)  long   0x2425b6229c8 static Fseq[s_dt]
+
+   ▌  stmtExpr  void  (top level) (IL 0x00E...  ???)
+   └──▌  return    void  
+
+```
+After liveness analysis the dead stores have been eliminated:
+```
+   ▌  stmtExpr  void  (top level) (IL 0x008...  ???)
+   └──▌  call help long   HELPER.CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE
+      ├──▌  const     long   0x7ffc8bbb4e10
+      └──▌  const     int    1
+
+   ▌  stmtExpr  void  (top level) (IL 0x008...  ???)
+   │  ┌──▌  const     struct 0
+   └──▌  =         struct (init)
+      └──▌  obj(16)   struct
+         │  ┌──▌  const     long   8 Fseq[#FirstElem]
+         └──▌  +         byref 
+            └──▌  indir     ref   
+               └──▌  const(h)  long   0x2425b6229c8 static Fseq[s_dt]
+
+   ▌  stmtExpr  void  (top level) (IL 0x00E...  ???)
+   └──▌  return    void  
+```
+And here is the resulting code, going from a code size of 129 bytes down to 58.
+```
+sub      rsp, 40
+mov      rcx, 0x7FFC8BBB4E10
+mov      edx, 1
+call     CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE
+xor      rax, rax
+mov      rdx, 0x2425B6229C8
+mov      rdx, gword ptr [rdx]
+add      rdx, 8
+vxorpd   ymm0, ymm0
+vmovdqu  qword ptr [rdx], ymm0
+add      rsp, 40
+ret 
+```
+
+Work Items
+----------
+This is a preliminary breakdown of the work into somewhat separable tasks. Those whose descriptions
+are prefaced by '*' have been prototyped in an earlier version of the JIT, and that work is now
+being re-integrated and tested, but may require some cleanup and/or phasing with other work items
+before a PR is submitted.
+
+  ### Mostly-Independent work items
+1.	*Replace block ops with assignments & new nodes.
+
+2.	*Add new fixed-size types, and normalize them in the importer (might be best to do this with or after #1, but not really dependent)
+
+3.	LSRA
+    * Enable support for multiple destination regs, for multi-reg call nodes (for x64/ux)
+    * Handle multiple destination regs for ldp (could be done before or concurrently with the above)
```

Comment:
(Minor) offline Carol has clarified this is a ldp optimization for 16-byte structs while passing as params on Arm.  It would be a good idea to be explicit to not confuse with a generic ldp optimization for Arm.